ocaml/xapi/xapi_network.ml

```diff
@@ -188,7 +188,7 @@ let stem = "xapi"
 let pool_introduce ~__context ~name_label ~name_description ~mTU ~other_config ~bridge ~managed =
   let r = Ref.make() and uuid = Uuid.make_uuid() in
   Db.Network.create ~__context ~ref:r ~uuid:(Uuid.to_string uuid)
-    ~current_operations:[] ~allowed_operations:[]
+    ~current_operations:[] ~allowed_operations:[] ~purposes:[]
```

Comment:
Same comment as below about preserving `purposes` after pool join - we should decide whether we want to do that.

Reply:
Yes, we should, so we should add a parameter to `network.pool_introduce`.